drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/enrollment_attribute_calculations.rb

```diff
@@ -116,66 +116,48 @@ def race_list(enrollment)
       end
 
       def ce_infos_for_batch(filter, batch)
-        household_enrollments = household_enrollments(filter, batch)
-        ce_events = ce_events(filter, household_enrollments)
+        # Find the active enrollments in the CE Project set for the HoH of the enrollments in the batch
+        ce_project_enrollments = GrdaWarehouse::ServiceHistoryEnrollment.
+          entry.
+          where(project_id: GrdaWarehouse::Hud::Project.where(id: filter.secondary_project_ids).pluck(:project_id)).
+          open_between(start_date: filter.start_date, end_date: filter.end_date).
+          where(id: batch.joins(:service_history_enrollment_for_head_of_household).select(:id))
+
+        enrollments_by_hoh = ce_project_enrollments.
+          preload(:client).
+          group_by { |enrollment| [enrollment.client.personal_id, enrollment.data_source_id] }
+
+        # Find the events for HoHs associated with the CE project enrollments above.
+        # Due to the possibility of finding enrollments with the same id from other data sources, this may pull
+        # more events than required, but they will end up in unused groups.
+        ce_events = GrdaWarehouse::Hud::Event.
+          where(enrollment_id: ce_project_enrollments.pluck(:enrollment_group_id), event: SERVICE_CODE_ID.keys, event_date: filter.range).
+          order(event_date: :asc).
+          group_by { |event| [event.personal_id, event.data_source_id] }
```

Comment:
Same here, why not join `client: :warehouse_client_source` and pluck `:destination_id` ?